### ISSUE TITLE
[Android Bootcamp] Project 2: Grid Image Search - Ready for Review

### ISSUE THREAD

**Issue by vlaljani** [vlaljani/GridImageSearch#1]
Hi, 

My app is complete. Please review. /cc @codepathreview @codepath

Thanks much! 

Best,
Vibha

**Comment by codepathreview** [comment]
:+1: Excellent submission again, Vibha! Very clean UI and code, thanks for putting in so much effort. Few notes:
- Good to see you used a filters object to represent the filters to apply to the query
- Consider using `Parcelable` instead of `Serializable` for [ImageFilters](https://github.com/vlaljani/GridImageSearch/blob/master/app/src/main/java/com/codepath/gridimagesearch/models/ImageFilters.java)
- Properly used spinners (with populated values) to allow users to select filters
- Nice effort on improving the UI/UX
- Nice job adding the share intent to the image detail view!
- Nice to see you added the search box to the ActionBar!
- Consider creating a static method in a Helper/Utility class for common methods such as checking for internet
- Consider adding [progress bars](http://guides.codepath.com/android/Handling-ProgressBars#actionbar-progress-bar) to your app whenever a network request goes out in addition to the placeholder images.

Also, here's a more detailed [Project 2 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_2_feedback) which covers the most common issues with this submitted project. Read through the [feedback guide point-by-point](http://courses.codepath.com/snippets/intro_to_android/project_2_feedback) to determine how you could improve your submission. 

Let us know if you have any other thoughts or questions about this assignment. The next assignment (Twitter Client) will be especially important since it introduces the majority of the remaining pieces necessary to build a fully functional API client with complex feeds of data and user creation.

**Comment by vlaljani** [comment]
Thank you for the feedback and guide! :) I'll keep those in mind for the next assignment. Btw, I did create and use ImageFiltersParcelable for Image Filters. Was that not done properly / as expected?

**Comment by codepathreview** [comment]
May be you forgot to push it? Because i still see your model objects implementing Serializable class and not [Parcelable](http://developer.android.com/reference/android/os/Parcelable.html)

**Comment by vlaljani** [comment]
Oh actually that ImageFilters class is not used anywhere. I just took all its methods and fields and put it in ImageFiltersParcelable and used that directly. I guess that's not the best way to go about it?

**Comment by codepathreview** [comment]
Yes that's the correct way.. Maybe it's just in the wrong package. Should be in the `model` package. May be you can refactor it later and remove unused files as it creates unnecessary confusion :)
